src/back/auth/src/api/login.cpp

```diff
@@ -17,20 +17,25 @@ Login::Login(
 	const components::ComponentContext& ctx)
 	: server::handlers::HttpHandlerBase{conf, ctx}
 	, _s{ctx.FindComponent<Service>()}
+	, _mapHttpMethodToSchema{LoadSchemas(kName, _s.GetJSONSchemasPath())}
 {}
 
 std::string Login::HandleRequestThrow(
 	const server::http::HttpRequest& req,
 	server::request::RequestContext&) const
 {
-	auto redirectPath = req.GetArg("redirectPath");
-	if (redirectPath.empty())
-	{
-		redirectPath = req.GetHeader(http::headers::kReferer);
+	const formats::json::Value body;
+	const auto params = ValidateRequest(_mapHttpMethodToSchema, req, body);
+
+	std::string redirectPath;
+	if (!params.HasMember("redirectPath")) {
+		redirectPath = params[http::headers::kReferer].As<std::string>();
 		redirectPath = http::ExtractPath(redirectPath);
+	} else {
+		redirectPath = params["redirectPath"].As<std::string>();
 	}
```

Comment:
```suggestion
	auto redirectPath = params["redirectPath"]
		.As<std::string>(formats::json::Value::DefaultConstructed{});
	if (redirectPath.empty()) {
		redirectPath = params[http::headers::kReferer].As<std::string>();
		redirectPath = http::ExtractPath(redirectPath);
	}
```

Проверь пожалуйста, может сработает более короткий вариант:
```suggestion
	auto redirectPath = params["redirectPath"].As<std::string>({});
	if (redirectPath.empty()) {
		redirectPath = params[http::headers::kReferer].As<std::string>();
		redirectPath = http::ExtractPath(redirectPath);
	}
```

Reply:
Вот такой вариант сработал:
```
    auto redirectPath = params["redirectPath"]
		.As<std::string>(formats::json::Value::DefaultConstructed{});
	if (redirectPath.empty()) {
		redirectPath = params[http::headers::kReferer].As<std::string>();
		redirectPath = http::ExtractPath(redirectPath);
	}
```

Reply:
Более короткий не работает, не может найти параметр, если его нет и если не указать дефолтное значение As<std::string>(formats::json::Value::DefaultConstructed{});